R/realtime.service.R

```diff
@@ -0,0 +1,182 @@
+#'@title create summary json for an individual storm item
+#'@description This service parses the metadata record and creates a summary specific to the STORM theme, 
+#'with subtype being defined by the attribute name
+#'@param serviceEndpoint valid xml, either local or a url
+#'@param attribute an attribute to use form the shapefile corresponding to 
+#'\code{serviceEndpoint}
+#'@return Serialized JSON for summary
+#'@importFrom jsonlite toJSON
+#'@import XML
+#'@examples
+#'serviceEndpoint  <-	'http://olga.er.usgs.gov/data/NACCH/GOM_erosion_hazards_metadata.xml'
+#'attribute	<-	'PCOL'
+#'summary	<-	realtime.service(serviceEndpoint,attribute)
+#'print(summary)
+#'@export
+realtime.service = function(serviceEndpoint,attribute=NULL){
+	subType	<-	tolower(attribute)
+
+	subTypeDataSrc <- names(sourceSynonyms)
+
+	doc <- xmlInternalTreeParse(serviceEndpoint)
+	title <- xmlValue(getNodeSet(doc,'//citation/citeinfo/title')[[1]])
+  titleParts <- extractInfoFromTitle(title)
+  
+	abstract	<-	xmlValue(getNodeSet(doc,'//descript/abstract')[[1]])
+	dataSrc <-  sapply(getNodeSet(doc,'//dataqual/lineage/srcinfo/srccite/citeinfo/title'),xmlValue)
+  sourceContent <-  paste(dataSrc,collapse='|')
+  
+  # NULL if it is for the whole storm
+  if (length(subType) == 0) {
+    tiny.text <- paste(titleParts$name, "Assessment of Potential Coastal-Change Impacts: NHC Adv.", titleParts$advNum)
+    medium.title <- titleParts$name
+    medium.text <- paste0("Potential coastal change impacts during a direct landfall of ",
+                         titleParts$name, ": ", titleParts$advFull)
+    full.title <- title
+    full.text <- paste("This dataset contains a coastal erosion hazards analysis for",
+                        paste0(titleParts$name, "."), "The analysis is based on a storm-impact scaling model",
+                        "that combines observations of beach morphology with hydrodynamic models to predict how sandy beaches,",
+                        "the first line of defense for many coasts exposed to tropical storms and hurricanes, will respond during",
+                        "a direct landfall. Storm-induced total water levels, due to both surge and waves, are compared to beach",
+                        "and dune elevations to determine the probabilities of three types of coastal change - collision (dune erosion),",
+                        "overwash, and inundation.")
+    full.text <- paste(full.text, getSurgeDescription(doc))
+    full.text <- paste(full.text, "Maximum wave heights in 20-m water depth, obtained from the NOAA WaveWatch3",
+                       "model 7-day forecast, were used to compute wave runup elevations at the shoreline.",
```

Comment:
I wonder if it's just a mistake on her part, check the overwash paragraph.. 'Maximum wave heights in 20-m water depth, obtained from the NOAA WaveWatch3 model 7-day forecast, were used to compute wave runup elevation at the shoreline. Dune elevations were extracted from lidar surveys. '